### ISSUE TITLE
Bug while starting my qterminal in a mounted filesystem

### ISSUE THREAD

**Issue by KZ25T** [lxqt/qterminal#1064]
Apologize at first
I am not a native English speaker and please forgive me for my mistakes in English. And it's my first time to report a bug on Github Issue, many mistakes may happen even if I have read some of the manuals and illustrations but not all. If it's useless, please forgive me and close it.

## Expected and current Behavior

I'm using qterminal in my kali linux:

```bash
$ uname -sr
Linux 6.3.0-kali1-amd64
$ qterminal --version
1.2.0
```

and my desktop is xfce4.

If a filesystem is mounted like ext4, ntfs or fat16 on `/mnt/mountpoint`, then I open that path in thunar (file and folder manager in xfce) and start qterminal by pushing the right button of the mouse and click `open terminal here`, then the begining path of my shell is `/mnt/mountpoint`. Then I close my thunar and command `cd ..` in qterminal to leave that mounted filesystem and unmount it:

```bash
$ sudo unmount /mnt/mountpoint
$ umount: /mnt/mountpoint: target is busy.
```

I can only unmount it when I close my qterminal and start a new one. What I expected is to be able to unmount when my just changed my working path to leave the filesystem, even if my qterminal started firstly in that filesystem.

I have not read the source code in this repository and I cannot give a way to fix it. I will be appreciate if someone can fix it.

## System Information
* QTerminal version: 1.2.0
* Distribution & Version: kali linux
* Kernel:  6.3.0-kali1-amd64
* Qt Version: qt5 1.5

**Comment by stefonarch** [comment]
Confirming here in arch with QTerminal 1.3.
Whatever directory is switch to in this terminal the mount point isn't freed.

**Comment by stefonarch** [comment]
I noticed that also any new tab opened will open in the mounted directory (unchecked "open in current directory") as it would be the home dir where it usually opens with this setting unchecked.

**Comment by tsujan** [comment]
My wild guess is that, due to a bad design, the initial working directory remains with the shell process “forever”, until the process is finished by closing its corresponding tab or window.

**Comment by stefonarch** [comment]
Maybe there's nothing to do about it: xterm, kitty and konsole have the same issue.

**Comment by tsujan** [comment]
> Maybe there's nothing to do about it

Yes, I checked the code and didn't find the cause in it. Perhaps it's how the shell process works.

**Comment by luebking** [comment]
The file descriptor is kept open by qterminal, not the shell (my friendly disk handling script tells me what's blocking the umount ;))

```qterminal -w ~``` affects the PWD of the initial shell but not qterminal
QDir::setCurrent(QDir::homePath()) would help but that's not common behavior as everyone figured and you're overriding the users (implicit) choice of the process $PWD
Arguably the cli switch could be taken to also apply to qterminal itself, though that would not help @KZ25T 


Oh, look: https://github.com/lxqt/qterminal/blob/98c174251f8bddac8c8c4fc7758f1cda3a20334b/src/main.cpp#L150
